src/poprox_recommender/recommenders/configurations/nrms.py

```diff
@@ -44,11 +45,23 @@ def configure(builder: PipelineBuilder, num_slots: int, device: str):
     n_scorer = builder.add_component("scorer", ArticleScorer, candidate_articles=e_candidates, interest_profile=e_user)
     n_ranker = builder.add_component("ranker", TopkRanker, {"num_slots": num_slots}, candidate_articles=n_scorer)
 
+    # Image selection
+    caption_embedder = CaptionEmbedder(
+        CaptionEmbedderConfig(model_path=model_file_path("nrms-mind/news_encoder.safetensors"), device=device)
+    )
+    image_selector = ImageSelector(caption_embedder)
+    o_with_images = builder.add_component(
+        "image-selector",
+        image_selector,
+        recommendations=n_ranker,
+        interest_profile=e_user,
+    )
+
     # Fallback: sample from user topic interests
     n_topic_filter = builder.add_component(
         "topic-filter", TopicFilter, candidate=i_candidates, interest_profile=i_profile
     )
     n_sampler = builder.add_component("sampler", UniformSampler, candidates1=n_topic_filter, candidates2=i_candidates)
 
     # Combine primary ranker and fallback
-    builder.add_component("recommender", FillRecs, {"num_slots": num_slots}, recs1=n_ranker, recs2=n_sampler)
+    builder.add_component("recommender", FillRecs, {"num_slots": num_slots}, recs1=o_with_images, recs2=n_sampler)
```

Comment:
I'd probably put the image selector after the fill to make sure all images are personalized regardless of where the selected items come from:
```suggestion
    n_fill = builder.add_component("fill", FillRecs, {"num_slots": num_slots}, recs1=n_ranker, recs2=n_sampler)
    n_images = builder.add_component(
        "recommender",
        image_selector,
        recommendations=n_fill,
        interest_profile=e_user,
    )
```
By convention, the last node in the pipeline is called `recommender`, which helps us find the right output from among several possibilities.